### ISSUE TITLE
Link sanity checking

### ISSUE THREAD

**Issue by sgstair** [ludumdare/ludumdare#1283]
When users post a link to their game, let's sanity check it and provide a mechanism to get feedback if it's not suitable or broken. This was partly discussed in #1197 

Core necessary features:
* If server has a link handler for a link type, use it to get data about the link.
* If not, HEAD request the URL to confirm it's not 404
* If there are problems (link is 404, or otherwise the link system flags problems), report back to user editing their game page by adding a warning box by the link informing them of the deadness of the link. (already partly implemented for #1197, just needs to be extended for this case)

Implementation:
* Probably add an API that requires login which will accept a URL, confirm its suitability and respond positively or with an error message. May want to add rate limiting

**Comment by local-minimum** [comment]
Just mention https://github.com/ludumdare/ludumdare/pull/1025 again. Think we shoul either use some of what is there for this or scrap it

**Comment by sgstair** [comment]
yeah, that is the plan.

**Comment by povrazor** [comment]
> If not, HEAD request the URL to confirm it's not 404

Oh hey! Wacky idea: Can we perform HEAD requests with JS? I.e. can this be done client side?

Initially my idea here had the server walking and periodically checking all known links. We might want to do this anyway, that way we can notify users when their links go bad. 

In the grand scheme of things, HEAD requests are pretty cheap. I'm now wondering if we could make the client do the validity check when they visit a page?

**Comment by sgstair** [comment]
We possibly shouldn't do it client side, the problem is that the user has access to the site (due to being signed in) and doesn't realize it's 404 to other people.

**Comment by povrazor** [comment]
Good point, but I believe you can specifically decide if credentials are included in XHR's/Fetches.
